temporal-sdk/src/test/java/io/temporal/internal/worker/WorkflowSlotTests.java

```diff
@@ -49,9 +54,16 @@ public class WorkflowSlotTests {
   private final int MAX_CONCURRENT_WORKFLOW_TASK_EXECUTION_SIZE = 100;
   private final int MAX_CONCURRENT_ACTIVITY_EXECUTION_SIZE = 1000;
   private final int MAX_CONCURRENT_LOCAL_ACTIVITY_EXECUTION_SIZE = 10000;
+  private final CountingSlotSupplier<WorkflowSlotInfo> workflowTaskSlotSupplier =
```

Comment:
I'm not sure how I feel about this because now you are not longer testing the default worker setting.

Reply:
It's a fairly thin wrapper and the only reasonable way I had to actually test the slot mechanism directly instead of via metrics without exposing a bunch more stuff publicly. 

If you have another suggestion I'm open for sure, but, this was seemingly the best way

Reply:
Yeah I was thinking you could have a few tests that schedule more activities/workflows then you have slots so you know some had to be freed. Thoughts on a test like that? You can also keep a static count in the activity to cross check with.

Reply:
Sure I can add something like that